File: user_spring/src/main/java/com/kjj/user/dto/info/ManagerInfoDto.java
package com.kjj.user.dto.info;

import com.kjj.user.entity.User;
import lombok.AllArgsConstructor;
import lombok.Getter;
import lombok.NoArgsConstructor;

@Getter
@AllArgsConstructor
@NoArgsConstructor
public class ManagerInfoDto {
    private Long id;
    private String loginId;

    public static ManagerInfoDto from(User manager) {
        return new ManagerInfoDto(
                manager.getId(),
                manager.getUsername()
        );
    }
}

File: user/src/main/kotlin/com/kjj/user/dto/info/ManagerInfoDto.kt
package com.kjj.user.dto.info

import com.kjj.user.entity.User

data class ManagerInfoDto (
    val id: Long,
    val username: String
){
    // java의 static method
    companion object {
        fun from(manager: User): ManagerInfoDto {
            return ManagerInfoDto(
                id = manager.id!!,
                username = manager.username
            )
        }
    }
}